app/components/appMainModule/committeePageModule/committeePageRoutes.js

```diff
@@ -8,19 +8,23 @@
 'use strict';
 
 function committeePageRoutes($stateProvider) {
-//module.exports = function($stateProvider) {
   $stateProvider.state({
     name: 'appMain.committee',
-    // name: 'appMain.locality.committee',
     abstract: true,
-    url: '^/committee/:committee_id',
+    url: '^/committee/:filer_id',
     template: '<ui-view></ui-view>',
     controller: 'committeePageController',
     resolve: {
-      committee: function($stateParams, disclosureApi) {
-        return disclosureApi.committee.get({
-          committee_id: $stateParams.committee_id
+      committee: function($stateParams, static_api) {
+        return static_api.committee.get({
+          filer_id: $stateParams.filer_id
+        });
+      },
+      contributions: function($stateParams, static_api) {
+        var apiCall = static_api.committee.contributions({
+          filer_id: $stateParams.filer_id
         });
+        return apiCall.$promise;
```

Comment:
Yes! This is supposed to be the way `resolve` works. It waits for the promises to resolve before running the route controller. I keep forgetting that `ngResource` doesn't return a promise, it returns a resource object with a `$promise` property. We should eventually update all of our `resolve`s to this and it will help with the rendering templates before the data is ready which causes some flicker.